### ISSUE TITLE
Official declaration of "I'm done" and shutdown of crowd bypass service

### ISSUE THREAD

**Issue by Sainan** [Sainan/Universal-Bypass#1951]
I think in the age of advertisements, an adblocker is a necessity for browsing the web, but over time the content and ads have merged more and more together, sparking the need for new solutions such as SponsorBlock and Universal Bypass. However, the latter solution here sucks, as I only ever encounter Linkvertise and Boost.ink — neither of which are covered due to the kind of society we found ourselves in. To make matters even worse, the extension was removed from the monopolistic Chrome Web Store, making the installation simply too annoying for the majority of web users. Adding these factors together makes me really not want to maintain this software. However, it is also eating up resources for the crowd bypass service to operate, and while I do think crowd bypass was a good ace up our sleves for this neverending cat & mouse game, it simply didn't address some issues, and not enough time was put into properly integrating it with some websites, so I have decided to pull the plug. Other than that, I will keep this extension available as-is. As a final note, this software is free of copyright encumberments, so I hope someone will make a better alternative or a continuation that will actually do the job it's supposed to do.

**Comment by Sainan** [comment]
Took a moment, but here's an export of the crowd bypass database in CSV format at https://universal-bypass.org/links_export.csv with the keys being `link, target, reports, created`

**Comment by pras92** [comment]
I understand the hardships in maintaining this project, especially with the need to spend for resources to keep UB operating. While I'm a pretty new user of UB, I thank you for keeping this up all this while.

When do you plan to pull the plug? Can you give us an idea of the resources required to operate the current/final version of UB?

Even if there's no further bug fixes or additional features, the UB available today would prove to be very much useful to many, I'm sure.

**Comment by Sainan** [comment]
As I've said, I'll keep the extension available as it is, since that's no hassle for me to do, but I've pulled the plug on the crowd bypass service, so that's a few websites that are now no longer bypassed.

**Comment by Ciccio90x** [comment]
This was one of the few REALLY USEFUL extensions. I am very sorry that you will not continue to support this extension ...

Why don't you add an option where people can upload their crowdfunding lists so you don't have to keep the database yourself?

**Comment by Sainan** [comment]
Anyone interested and able is free to add all the functionality they want. ;)

**Comment by Ciccio90x** [comment]
Can u reupload your database on another site ? After 1,5GB the download fail

**Comment by pras92** [comment]
> Can u reupload your database on another site ? After 1,5GB the download fail

Works fine for me. It's 2.09GB in size. Try using a download manager, and you can even resume should if fail, since it's a static link.

**Comment by fatihdemirbal** [comment]
> > Can u reupload your database on another site ? After 1,5GB the download fail
> 
> Works fine for me. It's 2.09GB in size. Try using a download manager, and you can even resume should if fail, since it's a static link.

how we using?

**Comment by darkpoet78** [comment]
> how we using?

Use Wget.  Works well.

**Comment by liam-maps** [comment]
>  To make matters even worse, the extension got removed from the monopolistic Chrome Web Store, making the installation very annoying for the majority of web users. 

Not using Chrome is a good start.

**Comment by toyboatcash** [comment]
im not sure if this would work can we use something like https://zeronet.io/ with option files for the crowd bypass service as a replacement. proxy for zeronet https://zero.acelewis.com.

**Comment by chimil** [comment]
one of the few useful and necessary extensions in browsers, is really unfortunate that you cannot continue to support it but if necessary I can help by donating space in the cloud I am a zoolz.com business user, I can donate the space in the cloud that you need

**Comment by marcussacana** [comment]
@Sainan is possible to we get the universall bypass database of links or a documented api to use the api as well?
Download the database should be better since we can't trust a project in his end of life to keep their servers running for a long term time

**Comment by Hunam6** [comment]
I created a few mirrors just in case:
https://mega.nz/file/fJ5hQYab#ItbsiesPvqEY1alT-jIqw6F6Ny3I7RUJ6Z693_50Mco
https://www.mediafire.com/file/kfsn70v9k705aom/links_export.csv/file
https://icedrive.net/0/702prAhFpP
https://multiup.org/en/mirror/fc1b1e63acb3d4156fdb610c58f911e6/links_export.csv

**Comment by marcussacana** [comment]
@Sainan can you make the server-side code open source?

**Comment by TheCatHacker562** [comment]
@Sainan make the server open source

**Comment by Sainan** [comment]
If you can't imagine what the server code looks like, I doubt you'll go far in maintaining a project like this. 🙃

**Comment by marcussacana** [comment]
who know, maybe we are just lazy?

**Comment by Hunam6** [comment]
@Sainan What is so important in this server that you don't want to release it?

**Comment by bingoxo** [comment]
is anyone maintaining a fork of this project ?

**Comment by marcussacana** [comment]
I think not yet

**Comment by TheCatHacker562** [comment]
Yes there is one

**Comment by marcussacana** [comment]
What fork?

**Comment by TheCatHacker562** [comment]
sorry for the delay
https://github.com/StuffNoOneCaresAbout/Universal-Bypass

**Comment by marcussacana** [comment]
> sorry for the delay
> https://github.com/StuffNoOneCaresAbout/Universal-Bypass

This repo don't looks active 3 PR ignored,
the you can see a new update but seens to be just a automatic periodic commit time update without any changes at all.
No idea if the owner did that for some reason, maybe he is implementing the server-side of the things, since Sainan don't wanna make the things easy to us.

**Comment by TheCatHacker562** [comment]
Idk I just found it
It says "auto update" I guess that is working on updating the thing?

**Comment by confusedlad** [comment]
Can anyone please help me, i wanna use the crowd sourced export, i downloaded the .csv file, what do I do next so that my extension works with those sites? Need Help

**Comment by TheCatHacker562** [comment]
I think that you can't, I don't know

**Comment by confusedlad** [comment]
I mean i should at least be able to store bypass definition which i have already used, think of it like crowd sourcing definitions but only i source them and its not server dependent or rather offline managed...? Anyone with a idea on how can this work out, please help me out?

**Comment by confusedlad** [comment]
I mean i should at least be able to store bypass definition which i have already used, think of it like crowd sourcing definitions but only i source them (so it actually dosent make it crowd sourced, ok self sourced) and its not server dependent or rather offline managed...? Anyone with a idea on how can this work out, please help me out?

**Comment by Beaness** [comment]
@Sainan are you still going to accept prs to the repository?

**Comment by Sainan** [comment]
@Beaness No, I'm not accepting any PRs. As I said, I have not much interest in maintaining this project anymore.

**Comment by BS-zombie** [comment]
I would just like to say thanks, @Sainan, for your work on this, its a really cool extension and I love it. I would, however, like to keep the crowd-bypass feature going, and continue to create new bypasses, so if anyone is interested then join a discord I made and we can keep this extension going, and maybe even create new features (and maybe do something about linkvertize...). https://discord.gg/RSAf7b5njt

**Comment by phanirithvij** [comment]
> Took a moment, but an export of the crowd bypass database in CSV format is now available at https://universal-bypass.org/links_export.csv with the keys being `link, target, reports, created`

@Sainan I'm assuming the extension has received a lot more links as of today (especially me I clicked 100s of first-time links which are now not in the csv file)
Can you please update this or better yet provide a daily/weekly update github workflow?

**Comment by BS-zombie** [comment]
No, unfortunately, she can't. The server was shut down, and no longer collects or gives links. We are working on restoring this in FastForward, if you are interested.

https://discord.gg/RSAf7b5njt
https://github.com/FastForwardTeam/FastForward

**Comment by LightWayUp** [comment]
I just found out about the discontinuation of this project, and would like to thank @Sainan for the awesome work over the years, as well as everyone who has contributed.

Regarding the change, I find it a little difficult to be aware, as there were no notifications or pop ups about this, and the extension options page and even the description on [Firefox Browser Add-ons](https://addons.mozilla.org/en-US/firefox/addon/universal-bypass/) don't mention anywhere that the project is now unmaintained. This means thousands of existing users are potentially running the (increasingly) outdated code of the extension.

Should a final update be made to the extension? I see that [the notice on the website and in README.md](https://github.com/Sainan/Universal-Bypass/commit/ff4fbc4) were added some while ago, and propose to add the notice to the options page too:

[html/options.html](https://github.com/Sainan/Universal-Bypass/blob/master/html/options.html)
```patch
--- html/options.html    (ff4fbc4)
+++ html/options.html    (HEAD)
@@ -29,4 +29,7 @@
 		<div id="counter" class="uk-alert uk-alert-primary uk-hidden">
 			<p><span></span> <a href="https://universal-bypass.org/support" target="_blank" data-message="support"></a></p>
 		</div>
+		<div class="uk-alert-warning uk-alert" uk-alert="">
+			<p>Universal Bypass is unmaintained. Check out <a href="https://fastforward.team/">FastForward</a>, which is a continuation of this project.</p>
+		</div>
 		<h2 data-message="options"></h2>
```

(In addition, pop ups can be shown to make the information much more prominent, however this is probably a terrible idea, especially if the extension is going to stay available):

[injection_script.js](https://github.com/Sainan/Universal-Bypass/blob/master/injection_script.js)
```patch
--- injection_script.js    (ff4fbc4)
+++ injection_script.js    (HEAD)
@@ -770,5 +770,6 @@
 ensureDomLoaded(()=>{
 	if(ignoreCrowdBypass)
 	{
 		persistHash("ignoreCrowdBypass")
 	}
+	insertInfoBox("Universal Bypass is unmaintained. Check out FastForward (https://fastforward.team/), which is a continuation of this project.")
```

It would be great if the suggestion can be considered, thanks!

**Comment by 1024mb** [comment]
I also vote for a final update making the user aware the add-on is discontinued now, more than half a year has gone and I'm finding about this just now.
